superset-frontend/src/components/IconButton/index.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { styled } from '@superset-ui/core';
+import Button from 'src/components/Button';
+import { ButtonProps as AntdButtonProps } from 'antd/lib/button';
+
+export interface IconButtonProps extends AntdButtonProps {
+  buttonText: string;
+  logo: string;
+  altText?: string;
+}
+
+const StyledButton = styled(Button)`
+  height: auto;
+  display: flex;
+  flex-direction: column;
+  padding: 0;
+`;
+const StyledImage = styled.div`
+  margin: ${({ theme }) => theme.gridUnit * 8}px 0;
+  padding: ${({ theme }) => theme.gridUnit * 4}px;
+
+  &:first-of-type {
+    margin-right: 0;
+  }
+
+  img {
+    width: fit-content;
+
+    &:first-of-type {
+      margin-right: 0;
+    }
+  }
+`;
+
+const StyledInner = styled.div`
+  max-height: calc(1.5em * 2);
+  overflow: hidden;
+  padding-right: 1rem;
+  position: relative;
+  white-space: break-spaces;
+
+  &::before {
+    content: '...';
+    inset-block-end: 0; /* "bottom" */
+    inset-inline-end: ${({ theme }) => theme.gridUnit * 2}px; /* "right" */
+    position: absolute;
+  }
+
+  &::after {
+    background-color: ${({ theme }) => theme.colors.grayscale.light4};
+    content: '';
+    height: 1rem;
+    inset-inline-end: ${({ theme }) => theme.gridUnit * 2}px; /* "right" */
```

Comment:
I think these can be hardcoded because they won't be affected by the theme.

Reply:
or at least we don't want them to be affected by the theme.

Reply:
Good call, that could get ugly later. Fixed at [`this commit`](https://github.com/apache/superset/pull/14818/commits/71495c8773b3482222ea3dae6e207c1e09f12835).